### PR TITLE
Feature/wfp 2274  display handover date and make sortable

### PR DESCRIPTION
**Pull request by gregkhawkins**
**This PR includes:**
- displaying the `handover-date` on the `Unallocated cases` page in the new `COM Handover Date` column
-  when we do not have a `handover-date`, displaying `N/A` on the `Unallocated cases page` in the new `COM Handover Date` column
- sorting for the new `COM Handover Date` column

**Details on sort:**
- I have done a big refactor to the sort as this was not fit for purpose for the table on the `Unallocated cases` page
- It was not fit for purpose because it does a `string sort` and in some of our columns (i.e. new `COM Handover Date` column and pre-existing `Initial appointment date` column) we convert the date to a formatted date string or `N/A` or `Unknown` in the case of appointment date)
- This means that `1 February 2024` would come before `1 January 2024` in an ascending search (as `F` comes before `J` in a string search)

**What is the situation in PROD right now?...**
- `Initial appointment date` sorting does not work as date is pre-formatted (e.g. `1 January 2024`)
- `Sentence Date` works fine as we do not pre-format the date. So at the point of sort the date value is in this format   `YYYY-MM-DD` (e.g. `2024-01-01`) and a `string sort` gets this right

**How does this PR fix this?...**
- It gives you the option of marking the table's column's header's metadata a new attribute called `data-type`
- When `data-type` is set to `DATE` then we convert the value to a date and then do a date sort
- if `data-type` is not set or is not set to `DATE` we do a string sort as always
- to use this data-type properly there are a lot of changes to our `mojSortableTable.js` script

### CODE REVIEW

File: integration_tests/integration/documents.spec.ts [hunk 1]
```diff
@@ -123,25 +125,37 @@ context('Documents', () => {
...
-        cy.get('table').within(() => cy.contains('button', heading).click())
-
-        cy.get('table').within(() => cy.contains('button', heading).should('have.attr', { 'aria-sort': 'descending' }))
-      })
-    })
+    const sortExpectations = [
+      {
+        columnHeaderName: 'Name',
+        orderedData: [
+          // todo: someone could look into this? Problem outlined....
```
Comment (AndySweilam) [inline integration_tests/integration/documents.spec.ts:132]: Can we get the upload date and use that instead 🤔

**Review by AndySweilam — APPROVED** [review]
Look good 👍

File: integration_tests/integration/documents.spec.ts (hunk 1 above)
Comment (gregkhawkins) [reply integration_tests/integration/documents.spec.ts:132]: this relates to the name column though. There is a sortable `Date created` column already

Comment (gregkhawkins) [reply integration_tests/integration/documents.spec.ts:132]: happy to discuss and figure out this columns sort though! But probs out of scope here I was just adding in coverage for this pre-existing sort and found this column should prob not be tested as the sort is broken (which is not a side effect of this PR's changes - has been broke for a while)

Comment (gregkhawkins) [reply integration_tests/integration/documents.spec.ts:132]: let me know if you want to discuss further :)

Comment (AndySweilam) [reply integration_tests/integration/documents.spec.ts:132]: Ok then looks fine then

**Pull request merged**